cargo-scout-lib/src/config/rust.rs

```diff
@@ -0,0 +1,112 @@
+use crate::config::Config;
+use serde::Deserialize;
+
+#[derive(Deserialize, PartialEq, Debug, Clone)]
+/// This struct represents a Cargo project configuration.
+pub struct CargoConfig {
+    members: Vec<String>,
+}
+
+impl Config for CargoConfig {
+    #[must_use]
+    fn get_members(&self) -> Vec<String> {
+        self.members.clone()
+    }
+}
+
+impl CargoConfig {
+    /// This function will instanciate a Config from a Cargo.toml path.
+    ///
+    /// If in a workspace, `get_members` will return the members
+    /// of the [[workspace]] members section in Cargo.toml.
+    ///
+    /// Else, it will return `vec![".".to_string()]`
+    ///
+    /// # cargo-scout-lib example
+    /// ```
+    /// # use cargo_scout_lib::config::Config;
+    /// # use cargo_scout_lib::config::rust::CargoConfig;
+    /// let config = CargoConfig::from_manifest_path("Cargo.toml")?;
+    /// // There is only one directory to lint, which is the current one.
+    /// assert_eq!(vec!["."], config.get_members());
+    /// # Ok::<(), cargo_scout_lib::error::Error>(())
+    /// ```
+    ///
+    /// # cargo-scout workspace example
+    /// ```
+    /// # use cargo_scout_lib::config::Config;
+    /// # use cargo_scout_lib::config::rust::CargoConfig;
+    /// let config = CargoConfig::from_manifest_path("../Cargo.toml")?;
+    /// // We will lint `./cargo-scout` and `./cargo-scout-lib`.
+    /// assert_eq!(vec!["cargo-scout".to_string(), "cargo-scout-lib".to_string()], config.get_members());
+    /// # Ok::<(), cargo_scout_lib::error::Error>(())
+    /// ```
+    pub fn from_manifest_path(p: impl AsRef<std::path::Path>) -> Result<Self, crate::error::Error> {
```

Comment:
As explained in [this pretty cool blog post](https://www.philipdaniels.com/blog/2019/rust-api-design/) AsRef<Path> allows the function to accept anything that allows a Path to be borrowed from it. So it allows us to automagically get a Path from a String, which I think is pretty cool !

Reply:
That's cool indeed!

I meant why don't we do
```
use std::path::Path;

... AsRef<Path>
```
Just for consistency with other cases

Reply:
Oh sorry I didn’t understand that’s what you meant!
Since it’s fairly easy to accomplish, and it’s not that critical, let’s mark it as a good first pull request, so someone new to rust and open source might be able to grab it !